trieste/utils/pareto.py

```diff
@@ -101,10 +108,155 @@ def _bounds_2d(front: TensorType) -> BoundedVolumes:
         )
 
         range_ = tf.range(len_front + 1)[:, None]
-        lower = tf.concat([range_, tf.zeros_like(range_)], axis=-1)
-        upper = tf.concat([range_ + 1, pf_ext_idx[::-1, 1:][: pf_ext_idx[-1, 0]]], axis=-1)
+        lower_result = tf.concat([range_, tf.zeros_like(range_)], axis=-1)
+        upper_result = tf.concat(
+            [range_ + 1, pseudo_front_idx[::-1, 1:][: pseudo_front_idx[-1, 0]]], axis=-1
+        )
+
+        return BoundedVolumes(lower_result, upper_result)
+
+    @staticmethod
+    def _is_test_required(smaller: TensorType) -> TensorType:
+        idx_dom_augm = tf.reduce_any(smaller, axis=1)
+        is_dom_augm = tf.reduce_all(idx_dom_augm)
+
+        return is_dom_augm
+
+    def _divide_conquer_nd(self, front: TensorType, threshold: TensorType = 0) -> TensorType:
```

Comment:
```suggestion
    def _divide_conquer_nd(self, front: TensorType, threshold: TensorType | float = 0) -> TensorType:
```
so the default makes sense?

Reply:
agree

Reply:
got this error
`ERROR test_pareto.py - TypeError: unsupported operand type(s) for |: '_GenericAlias' and 'type'`
will check it

Reply:
add `from __future__ import annotations` above all the other imports: this is an advanced mypy feature that requires this import

Reply:
okay 👌🏼